File: EduMS-Frontend/edums-frontend-DDD-System/src/app/modules/school-admin/interfaces/directorate-legal-case-log.ts
import type { CaseCategory, CaseStatus } from './common';

export interface DirectorateLegalCaseLog {
  id: number;
  directorateId: number;
  caseCodeNumber: string;
  caseCategory: CaseCategory;
  subjectTitle: string;
  involvedPartiesDescription: string;
  registrationDate: string;
  resolutionDate: string | null;
  caseStatus: CaseStatus;
  resolutionDecisionText: string | null;
  assignedLegalCounselEmployeeId: number | null;
  caseDocumentAttachmentUrl: string | null;
}

export type CreateDirectorateLegalCaseLogDto = Omit<DirectorateLegalCaseLog, 'id' | 'caseStatus'>;

export type UpdateDirectorateLegalCaseLogDto = Omit<DirectorateLegalCaseLog, 'directorateId' | 'caseStatus'>;

File: 01_Backend/EduMS.Backend/src/EduMS.Domain/Entities/M1_SchoolAdmin/DirectorateLegalCaseLog.cs
using System;
using EduMS.Domain.Common;

namespace EduMS.Domain.Entities;

public class DirectorateLegalCaseLog : BaseAuditableEntity
{
    public long DirectorateId { get; set; }
    public string CaseCodeNumber { get; set; } = string.Empty;
    public int CaseCategory { get; set; } // 1=LegalDispute, 2=AdministrativeComplaint, 3=RegulatoryInquiry, 4=Consultation
    public string SubjectTitle { get; set; } = string.Empty;
    public string InvolvedPartiesDescription { get; set; } = string.Empty;
    public DateTime RegistrationDate { get; set; } = DateTime.UtcNow;
    public DateTime? ResolutionDate { get; set; }
    public int CaseStatus { get; set; } // 1=UnderInvestigation, 2=InCourt, 3=ResolvedClosed, 4=ReferredToMinistry
    public string? ResolutionDecisionText { get; set; }
    public long? AssignedLegalCounselEmployeeId { get; set; }
    public string? CaseDocumentAttachmentUrl { get; set; }

    // Navigation Property
    public virtual Directorate? Directorate { get; set; }
    public virtual Employee? AssignedLegalCounselEmployee { get; set; }
}
